pynamodb/models.py

```diff
@@ -267,6 +264,18 @@ def __init__(self, name, bases, namespace, discriminator=None) -> None:
                 }
                 cls.DoesNotExist = type('DoesNotExist', (DoesNotExist, ), exception_attrs)
 
+    @staticmethod
+    def _initialize_indexes(cls):
+        """
+        Initialize indexes on the class.
+        """
+        cls._indexes = {}
```

Comment:
This changes current behavior where child classes now each have their own index dictionary instead of defaulting to the dictionary on the parent class.